bitcoin/src/blockdata/transaction.rs

```diff
@@ -1168,6 +1170,69 @@ impl From<&Transaction> for Wtxid {
     fn from(tx: &Transaction) -> Wtxid { tx.wtxid() }
 }
 
+/// Predicts the weight of a to-be-constructed transaction.
+///
+/// This function computes the weight of a transaction which is not fully known. All that is needed
+/// is an arbitrary set of TxIn and TxOut.
+///
+/// # Arguments
+///
+/// * `inputs` - a vector of inputs (TxIn).
+/// * `outputs` - a vector of outputs (TxOut).
+///
+/// # Usage
+///
+/// Determine the weight of a set of OutPoints (Utxos) before they are included in a transaction.
+/// For example, if we wish to calculate the possible fee for including a set of UTXOs in a
+/// transaction, this function can be used to predict the weight of a set of UTXOs prior to inclusion.
+///
+/// # Notes
+///
+/// This function uses the same internals as predict_weight, so the same caveats and assumptions
+/// apply to this function.
+pub fn predict_weight_with_inputs_outputs(inputs: &Vec<TxIn>, outputs: &Vec<TxOut>) -> Weight {
```

Comment:
In e4ae45c9f3504bcc0d46aed51547723d08d0149f:

You should never accept a `&Vec` as a function argument. In this case there is no need to accept a vector at all, since all you do with these inputs is iterate over them (and count them). So you might as well just accept an iterator like `predict_weight` does.

Reply:
Thanks for the feedback.  I'm curious why not ever accept a `&Vec`?  Since it's just a reference it should just be passing a pointer instead of copying the Vec.

Reply:
If the consumer has the items in some different data structure he needs to copy them into a vec first. If you accept a slice then it's possible to use also arrays and if you accept iterator then practically any usual data structure can be used, even deserializing from disk on-the-fly. So iterators are clearly the best if a function doesn't need random access.

Reply:
Ok, I see.  Thanks!